org.eclipse.xtext.tests/src/org/eclipse/xtext/resource/containers/LiveShadowedChunkedContainerTest.java

```diff
@@ -0,0 +1,265 @@
+import java.util.Map;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.xtext.naming.QualifiedName;
+import org.eclipse.xtext.resource.IEObjectDescription;
+import org.eclipse.xtext.resource.IResourceDescription;
+import org.eclipse.xtext.resource.LiveContainerTestLanguageInjectorProvider;
+import org.eclipse.xtext.resource.XtextResourceSet;
+import org.eclipse.xtext.resource.impl.ChunkedResourceDescriptions;
+import org.eclipse.xtext.resource.impl.LiveShadowedChunkedResourceDescriptions;
+import org.eclipse.xtext.resource.impl.ResourceDescriptionsData;
+import org.eclipse.xtext.resource.liveContainerTestLanguage.LiveContainerTestLanguagePackage;
+import org.eclipse.xtext.resource.liveContainerTestLanguage.Model;
+import org.eclipse.xtext.testing.InjectWith;
+import org.eclipse.xtext.testing.XtextRunner;
+import org.eclipse.xtext.testing.util.ParseHelper;
+import org.eclipse.xtext.workspace.ProjectConfigAdapter;
+import org.eclipse.xtext.workspace.WorkspaceConfig;
+import org.eclipse.xtext.xbase.lib.CollectionLiterals;
+import org.eclipse.xtext.xbase.lib.IterableExtensions;
+import org.eclipse.xtext.xbase.lib.ListExtensions;
+import org.eclipse.xtext.xbase.lib.Pair;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+
+/**
+ * @author koehnlein - Initial contribution and API
+ */
+@RunWith(XtextRunner.class)
+@InjectWith(LiveContainerTestLanguageInjectorProvider.class)
+public class LiveShadowedChunkedContainerTest {
+	@Inject
+	private ParseHelper<Model> parseHelper;
+
+	@Inject
+	private Provider<XtextResourceSet> resourceSetProvider;
+
+	@Inject
+	private IResourceDescription.Manager resourceDescriptionManager;
+
+	@Inject
+	private Provider<LiveShadowedChunkedResourceDescriptions> provider;
+
+	private WorkspaceConfig workspaceConfig;
+
+	private ProjectConfig fooProject;
+
+	private ProjectConfig barProject;
+
+	private URI fooURI;
+
+	private URI barURI;
+
+	private XtextResourceSet rs1;
+
+	private LiveShadowedChunkedContainer fooContainer;
+
+	private LiveShadowedChunkedContainer barContainer;
+
+	private LiveShadowedChunkedResourceDescriptions liveShadowedChunkedResourceDescriptions;
+
+	@Before
+	public void setUp() throws Exception {
+		workspaceConfig = new WorkspaceConfig();
+		fooProject = new ProjectConfig("foo", workspaceConfig);
+		barProject = new ProjectConfig("bar", workspaceConfig);
+		XtextResourceSet rs0 = resourceSetProvider.get();
+		fooURI = IterableExtensions.head(fooProject.getSourceFolders()).getPath().trimSegments(1)
+				.appendSegment("foo.livecontainertestlanguage");
+		barURI = IterableExtensions.head(barProject.getSourceFolders()).getPath().trimSegments(1)
+				.appendSegment("bar.livecontainertestlanguage");
+		Map<String, ResourceDescriptionsData> chunks = Collections.unmodifiableMap(CollectionLiterals.newHashMap(
```

Comment:
I'm not a big fan of this `newHashMap(Pair(..))` stuff... given this is only a test, I'm ok with it though.